src/main/java/com/google/sps/data/Chat.java

```diff
@@ -124,6 +123,25 @@ public static Chat chatExists(Datastore datastore, String participantId1, String
         }
     }
 
+    public static Chat newMessage(Datastore datastore, String chatId, String userId, String message) {
+        Key chatKey = datastore.newKeyFactory().setKind("chat").newKey(Long.parseLong(chatId));
+        Entity chatEntity = datastore.get(chatKey);
+
+        // Note: So far what I've found online says you cannot add to a list property 
+        // without getting the entire list and adding the new item to it, then replace 
+        // the old list property on the entity. There must be a better way but Im stumped right now
+        List<String> messages = chatEntity.getList("messages").stream().map(val -> ((StringValue) val).get()).collect(Collectors.toList());
+        messages.add(message);
+        // Now here I need to find out how to make a List<Value<?>> with all 
+        // the strings like above but in revers
+        
```

Comment:
[Example from our code](https://github.com/josueetcom/TeachMe/blob/8423bb4bcda1bbf748ac1b72f67a13ae51386b77/src/main/java/com/google/sps/servlets/UserServlet.java#L118-L119)

Though to be fair, you actually don't need to do the transformations if you just work with the `List<Value>` directly. E.g.

```java
List<Value> messages  = chatEntity.getList("messages");
messages.add(StringValue.of(message));
// Now store messages
```